dbfit-java/sqlserver/src/main/java/dbfit/environment/SqlServerEnvironment.java

```diff
@@ -80,15 +78,18 @@ protected String parseCommandText(String commandText) {
             String query) throws SQLException {
         DbParameterAccessorsMapBuilder params = new DbParameterAccessorsMapBuilder();
 
-        if (objname.contains(".")) {
-            String[] schemaAndName = objname.split("[\\.]", 2);
-            objname = "[" + schemaAndName[0] + "].[" + schemaAndName[1] + "]";
-        } else {
-            objname = "[" + normaliseName(objname) + "]";
+        objname = objname.replaceAll("[^a-zA-Z0-9_.#]", "");
+        String[] qualifiers = objname.split("\\.");
+        String bracketedName = "";
```

Comment:
Java Strings are immutable so when you do `+=`, a new String instance is created (the old ones are garbage collected). [`StringBuilder`](https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html) is intended to avoid that overhead to hopefully improve performance. `append` method and finally - `toString`.